hazelcast/src/main/java/com/hazelcast/spi/merge/KeyMergeDataHolder.java

```diff
@@ -14,21 +14,28 @@
  * limitations under the License.
  */
 
-package com.hazelcast.spi;
-
-import com.hazelcast.core.EntryView;
+package com.hazelcast.spi.merge;
 
 /**
- * Represents a read-only view of data structure entries for the merging process after a split-brain.
- * <p>
- * On key-less data structures like {@link com.hazelcast.core.ISet} or {@link com.hazelcast.ringbuffer.Ringbuffer},
- * the method {@link #getKey()} will return {@code null}.
- * <p>
- * Some methods return {@code -1} if statistics are not enabled or not implemented.
+ * Represents a read-only view of a data structure key/value-pair for the merging process after a split-brain.
  *
  * @param <K> the type of the key
  * @param <V> the type of the value
  * @since 3.10
  */
-public interface SplitBrainMergeEntryView<K, V> extends EntryView<K, V> {
+public interface KeyMergeDataHolder<K, V> extends MergeDataHolder<V> {
```

Comment:
minor : 
- maybe we can rename `KeyMergeDataHolder` as `MergingEntryHolder`(it has key&value)
- `MergeDataHolder` --> `MergingValueHolder`. Reasoning is when i see `MergeDataHolder`, i expect a `getData` method but we have `getValue`
- And for the others we can remove `Data` from the name `HitsDataHolder` --> `HitsHolder`
(These are from the biased seemed-intuitive perspective of mine)